backend/de.metas.business/src/main/sql/postgresql/ddl/functions/Reset_System_Base_Language.sql

```diff
@@ -0,0 +1,78 @@
+CREATE OR REPLACE FUNCTION Reset_System_Base_Language(p_source_language      varchar,
+                                                      p_destination_language varchar) RETURNS VOID
+AS
+$$
+DECLARE
+    destination_language  varchar := '';
+    source_language       varchar := '';
+    table_name            varchar;
+    base_table_id         numeric;
+    base_table            varchar;
+    query_set_template    text    := '';
+    final_update_query    text    := '';
+    v_columnname          varchar;
+    update_query_template varchar := 'WITH X AS (SELECT * FROM base_table_trl WHERE base_table_id = %s AND ad_language =  ''%s'') '
+                                         || 'UPDATE base_table as target '
+                                         || 'SET %s '
+                                         || 'FROM X '
+        || 'WHERE target.base_table_id = x.base_table_id ;' ;
+BEGIN
+
+    IF (p_source_language IS NULL OR TRIM(p_source_language) = '') THEN
+        RAISE WARNING 'No source language is provided';
+        RETURN;
+    END IF;
+
+    IF (p_destination_language IS NULL OR TRIM(p_destination_language) = '') THEN
+        RAISE WARNING 'No destination language is provided';
+        RETURN;
+    END IF;
+
+    source_language := TRIM(p_source_language);
+    destination_language := TRIM(p_destination_language);
+
+    -- update languages
+    UPDATE ad_language
+    SET isbaselanguage='N', issystemlanguage='Y'
+    WHERE ad_language = source_language;
+
+    UPDATE ad_language
+    SET isbaselanguage = 'Y', issystemlanguage = 'Y'
+    WHERE ad_language = destination_language;
+
+    -- Iterate over Trls tables
+    FOR table_name IN SELECT UPPER(tablename) FROM ad_table WHERE UPPER(tablename) LIKE UPPER('%_Trl') ORDER BY 1
+        LOOP
+            base_table := REPLACE(table_name, '_TRL', '');
```

Comment:
@Chemssou just to be on the safe side, i would do something like substring and take out last 4 chars.

because 

> REPLACE(table_name, '_TRL', '');

might work wrong for cases like `C_FLIGHT_CTRL_TRL`
(we don't have such a table now, but u got my point).